cent/client/async_client.py

```diff
@@ -0,0 +1,115 @@
+from typing import List, Optional, Any, Dict, TypeVar
+
+from cent.client.session import BaseAsyncSession, AiohttpSession
+from cent.methods import (
+    CentMethod,
+    BroadcastMethod,
+    PublishMethod,
+    SubscribeMethod,
+)
+from cent.types import (
+    SubscribeObject,
+    StreamPosition,
+    PublishObject,
+    BroadcastObject,
+    Override,
+)
+
+T = TypeVar("T")
+
+
+class AsyncClient:
+    def __init__(
+        self,
+        base_url: str,
+        api_key: str,
+        session: Optional[BaseAsyncSession] = None,
+    ) -> None:
+        """
+        :param base_url: Centrifuge base_url
+        :param api_key: Centrifuge API key
+        :param session: Custom Session instance
+        """
+
+        self._base_url = base_url
+        self.api_key = api_key
+        self.session = session or AiohttpSession(base_url=base_url)
+
+    async def publish(
+        self,
+        channel: str,
+        data: Dict[str, Any],
```

Comment:
Here data can be any JSON-encodable object - so probably use JSON type like in centrifuge-python (https://github.com/centrifugal/centrifuge-python/blob/a5c432885611077da10a5fda02a20f031ffc68b9/centrifuge/types.py#L5)?